### PR TITLE
state-delta: support redis dds

### PR DESCRIPTION
**Pull request by chengzhinei**
- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/okex/okexchain/blob/develop/CONTRIBUTING.md#pr-targeting))

- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Reviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/okex/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: libs/tendermint/delta/delta.go [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package delta
+
+import tmtypes "github.com/okex/exchain/libs/tendermint/types"
+
+type DeltaBroker interface {
+	SetBlock(block *tmtypes.Block) error
+	SetDelta(deltas *tmtypes.Deltas) error
```
Comment (chengzhinei) [inline libs/tendermint/delta/delta.go]: Deltas

File: libs/tendermint/types/deltas.go [hunk 2]
```diff
@@ -100,15 +112,13 @@ func GetCenterUrl() string {
 type Deltas struct {
 	ABCIRsp     []byte `json:"abci_rsp"`
 	DeltasBytes []byte `json:"deltas_bytes"`
+	WatchBytes	[]byte `json:"watch_bytes"`
```
Comment (chengzhinei) [inline libs/tendermint/types/deltas.go]: fmt

File: x/evm/watcher/types.go [hunk 3]
```diff
@@ -78,7 +78,7 @@ type WatchData struct {
 
 func (w *WatchData) Size() int {
 	if w == nil {
```
Comment (chengzhinei) [inline x/evm/watcher/types.go]: del

File: libs/tendermint/state/execution.go [hunk 4]
```diff
@@ -323,73 +298,134 @@ func (blockExec *BlockExecutor) ApplyBlock(
...
+	// not use delta, exe abci itself
+	if !applyDelta && !downloadDelta {
+		return false, deltas
 	}
-	if deltas != nil {
-		if deltaBytes, err = deltas.Marshal(); err != nil {
-			return
+
+	// get watchData and Delta from p2p
+	if len(deltas.ABCIRsp) >0 && len(deltas.DeltasBytes) > 0 {
```
Comment (chengzhinei) [inline libs/tendermint/state/execution.go]: if ApplyDelta()

**Review by ilovers — DISMISSED** [review]

**Review by xiangjianmeng — APPROVED** [review]

**Pull request merged**